### PR TITLE
message board unread messages

### LINKED ISSUE #1407: Country Message Board: Unread count
- [x] message board
- [] message board users

image

Related:

client/pages/AssessmentHome/FraHome/CountryMessageBoard/MessageBoard.tsx:22
client/pages/AssessmentHome/FraHome/CountryMessageBoard/MessageBoardUsers.tsx:71

### PR DESCRIPTION
**Pull request by rfontanarosa**
closes #1407

Schermata 2022-08-09 alle 15 43 00

### CODE REVIEW

File: client/pages/AssessmentHome/FraHome/CountryMessageBoard/MessageBoard.tsx [hunk 1]
```diff
@@ -57,9 +68,7 @@ const MessageBoard = () => {
               >
                 <Icon name="chat-46" className="icon-middle" />
                 {i18n.t<string>('landing.users.message')}
-                {countryMessageBoardUnreadMessages > 0 && (
-                  <div className="landing__user-message-count">{countryMessageBoardUnreadMessages}</div>
-                )}
+                {data.unreadMessages > 0 && <div className="landing__user-message-count">{data.unreadMessages}</div>}
```
Comment (sorja) [inline client/pages/AssessmentHome/FraHome/CountryMessageBoard/MessageBoard.tsx:71]: should we extract unreadMessages?

File: common/api/endpoint/ApiEndPoint.ts [hunk 2]
```diff
@@ -107,6 +107,10 @@ export const ApiEndPoint = {
       getMessage: () => apiPath('topic', 'message'),
       resolveTopic: () => apiPath('topic', 'resolve'),
     },
+    Stats: {
+      getUnreadChatMessages: () => apiPath('stats', 'getUnreadChatMessages'),
```
Comment (minotogna) [inline common/api/endpoint/ApiEndPoint.ts:111]: There should be only one endpoint and it's generic .

File: server/controller/messageCenter/index.ts [hunk 3]
```diff
@@ -10,4 +12,6 @@ export const MessageCenterController = {
   updateTopicReadTime,
   updateTopicStatus,
   markMessageDeleted,
+  getUnreadChatMessages: MessageTopicUserRepository.getUnreadChatMessages,
```
Comment (minotogna) [inline server/controller/messageCenter/index.ts:15]: Only one generic method

File: server/repository/assessmentCycle/messageTopicUser/index.ts [hunk 4]
```diff
@@ -11,5 +13,7 @@ export const MessageTopicUserRepository = {
   getOneOrNone,
   getReviewStatus,
   getReviewSummary,
```
Comment (minotogna) [inline server/repository/assessmentCycle/messageTopicUser/index.ts:15]: Only one generic method

**Review by minotogna — CHANGES_REQUESTED** [review]

File: server/api/messageCenter/getUnreadMessages.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import { MessageCenterController } from '@server/controller/messageCenter'
+import Requests from '@server/utils/requests'
+
+export const getUnreadMessages = async (req: Request, res: Response) => {
+  try {
+    const { countryIso, assessmentName, cycleName, type } = req.query as {
+      countryIso: CountryIso
+      assessmentName: AssessmentName
+      cycleName: string
+      type: MessageTopicType
```
Comment (minotogna) [inline server/api/messageCenter/getUnreadMessages.ts:17]: You need to pass the topic key and the response should return only one unread count ( only for the combination of assessment, cycle , countryIso , topicKey)

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**